python/lsst/cp/pipe/defects.py

```diff
@@ -919,11 +921,28 @@ class MergeDefectsCombinedTask(MergeDefectsTask):
     ConfigClass = MergeDefectsCombinedTaskConfig
     _DefaultName = "cpDefectMergeCombined"
 
+    @staticmethod
+    def chooseBest(self, inputs):
+        """Select the input with the most exposures used."""
+        best = 0
+        if len(inputs) > 1:
+            nInput = 0
+            for num, exp in enumerate(inputs):
+                # This technically overcounts by a factor of 3.
+                N = len([k for k, v in exp.getMetadata().toDict().items() if "CPP_INPUT_" in k])
+                if N > nInput:
+                    best = num
+                    nInput = N
+        return inputs[best]
+
     def runQuantum(self, butlerQC, inputRefs, outputRefs):
         inputs = butlerQC.get(inputRefs)
-        # Turn inputFlatDefects and inputDarkDefects into a list
-        # which is what MergeDefectsTask expects.
-        tempList = [inputs['inputFlatDefects'], inputs['inputDarkDefects']]
+        # Turn inputFlatDefects and inputDarkDefects into a list which
+        # is what MergeDefectsTask expects.  If there are multiple,
+        # use the one with the most inputs.
+        tempList = [self.chooseBest(inputs['inputFlatDefects']),
+                    self.chooseBest(inputs['inputDarkDefects'])]
```

Comment:
So the decision is to just use the one with the most inputs, rather than try to merge all of them from the different filters?  Have you confirmed that this does something sensible?

Reply:
In the {{cp_testing}} processing, all flats are generated prior to these defects, so the task needs to handle the existence of multiple inputs.  I didn't want to simply {{OR}} all of the inputs, as the verify ticket for combined defects showed that the dust spots can trigger masking if they're fainter than the threshold.  {{OR}}ing these together means that the output defects may have all faint areas in all dust spots in all filters contributing to the masks.
I will admit that choosing the one with the most inputs is completely arbitrary.  However, this is somewhat artificial problem; In general processing, the flats would be inspected prior to use, and a single optimal input selected.  Only in the {{cp_testing}} case does this decision need to be made automatically.